### PR TITLE
record audio node

### PR DESCRIPTION
**Pull request by jtydhr88**
BE changes for bounty task https://www.notion.so/drip-art/Audio-input-node-1706d73d365080eeb8e5e39790584dff?source=copy_link
and FE changes is here

- https://github.com/Comfy-Org/ComfyUI_frontend/pull/4289

### CODE REVIEW

File: comfy_extras/nodes_audio.py [hunk 1]
```diff
@@ -321,6 +321,24 @@ def VALIDATE_INPUTS(s, audio):
...
+class RecordAudio:
+    @classmethod
+    def INPUT_TYPES(s):
+        return {"required": {"audio": ("AUDIO_RECORD", {})}}
+
+    CATEGORY = "audio"
+
+    RETURN_TYPES = ("AUDIO", )
+    FUNCTION = "load"
+
```
Comment (guill) [inline comfy_extras/nodes_audio.py:333]: Note for any other reviewers (or future me): Because the uploaded files are named via timestamps by the frontend, an existing uploaded audio file should never be updated. That's why we don't need the `IS_CHANGED` here.

**Review by guill — APPROVED** [review]
Looks good to me

**Review by OrangeDoro — COMMENTED** [review]
Hi! I'm a grad student working on a research project about using large language models to automate code review. Based on your commit 236050c14843b25ea0cceec461b1a8de388ae589 and the changes in comfy_extras/nodes_audio.py, my tool generated this comment:
1. **Reinstate Input Directory Logic**: The lines that were removed for obtaining the input directory and filtering files based on their content types are crucial for ensuring that only valid audio or video files are processed. Consider reintroducing this logic to prevent potential errors later in the code when invalid file types are passed to the `load` method.
2. **Error Handling in `load` Method**: Implement error handling to manage cases where the input directory is invalid or when unsupported file types are encountered. This will help prevent runtime errors and provide clear feedback to the user.
3. **Input Validation**: Ensure that the `INPUT_TYPES` method validates the audio input provided. If the audio parameter is null or invalid, it could lead to runtime errors when attempting to process it in the `load` method.
4. **Exception Handling for `torchaudio.load`**: Wrap the call to `torchaudio.load` in a try-except block to handle exceptions that may arise from invalid files, such as unsupported formats or corrupted files.
5. **Testing for Validity**: Ensure that there are tests in place that validate the behavior of the `load` method when given various types of inputs, including invalid ones. This will help catch any issues that arise from the removal of the filtering logic.
6. **Edge Case Testing**: Ensure that tests cover edge cases, such as providing an invalid audio input or a non-existent file path, to ensure that appropriate exceptions are raised and handled gracefully.

As part of my research, I'm trying to understand how useful these comments are in real-world development. If you have a moment, I'd be super grateful if you could quickly reply to these two yes/no questions:


1. Does this comment provide suggestions from a dimension you hadn’t considered?

2. Do you find this comment helpful?

Thanks a lot for your time and feedback! And sorry again if this message is a bother.

**Comment by jtydhr88** [comment]
@OrangeDoro hi, the main issue of your LLM feedback is lacking some context, for this specific case, it could not only consider this single PR, as it also has FE part, the full feature required both BE (this) and FE, such as how we handle upload recoding to temp folder.
Regarding testing, due to we don’t add tests for this file right now, we may add tests later.

BTW, I don’t want to confuse the core team reviewer as we already approve this BE changes, and it is close to merge soon, could you remove your LLM feedback? if you want to share your study, you can email me directly with these information - wswdkt88@gmail.com
Thanks for understanding.

**Comment by jtydhr88** [comment]
@guill @comfyanonymous @christian-byrne As we already merge FE https://github.com/Comfy-Org/ComfyUI_frontend/pull/4289, please make sure we have this PR while include the latest FE

**Review by christian-byrne — APPROVED** [review]
This is good to merge now, as the frontend support has been added.
